session/pluginloader.go

```diff
@@ -112,13 +114,33 @@ func getUnixSocketDir(opts dgo.Map) string {
 	if v, ok := opts.Get("unixSocketDir").(dgo.String); ok {
 		return v.GoString()
 	}
-	if v := os.Getenv("TMPDIR"); v != "" {
+	if v := os.TempDir(); v != "" {
 		return v
 	}
 	return defaultUnixSocketDir
 }
 
-var defaultPluginTransport = "unix"
+func getDefaultPluginTransport() string {
+	if runtime.GOOS == "windows" {
+		k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.READ)
```

Comment:
I would very much like a comment describing what's going on here and why.

Reply:
Explanation added

Reply:
Perfect. Thanks!